From: Eric Van Hensbergen <ericvh@gmail.com>
Subject: [PATCH 005/111] net/9p: fix statsize inside twstat
Git-commit: 9d6939dac77102b09396ee0b89392ec7639612a7
Patch-Mainline: 2.6.33
References: FATE#311639

    stat structures contain a size prefix.  In our twstat messages
    we were including the size of the size prefix in the prefix, which is
    not what the protocol wants, and Inferno servers would complain.

	upstream commit: 9d6939dac77102b09396ee0b89392ec7639612a7

    Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>
Signed-off-by: Harsh Prateek Bora <harsh@linux.vnet.ibm.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 net/9p/client.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/net/9p/client.c b/net/9p/client.c
index 90a2eb9..a2e2d61 100644
--- a/net/9p/client.c
+++ b/net/9p/client.c
@@ -1222,10 +1222,11 @@ static int p9_client_statsize(struct p9_wstat *wst, int optional)
 {
 	int ret;
 
+	/* NOTE: size shouldn't include its own length */
 	/* size[2] type[2] dev[4] qid[13] */
 	/* mode[4] atime[4] mtime[4] length[8]*/
 	/* name[s] uid[s] gid[s] muid[s] */
-	ret = 2+2+4+13+4+4+4+8+2+2+2+2;
+	ret = 2+4+13+4+4+4+8+2+2+2+2;
 
 	if (wst->name)
 		ret += strlen(wst->name);
@@ -1266,7 +1267,7 @@ int p9_client_wstat(struct p9_fid *fid, struct p9_wstat *wst)
 		wst->name, wst->uid, wst->gid, wst->muid, wst->extension,
 		wst->n_uid, wst->n_gid, wst->n_muid);
 
-	req = p9_client_rpc(clnt, P9_TWSTAT, "dwS", fid->fid, wst->size, wst);
+	req = p9_client_rpc(clnt, P9_TWSTAT, "dwS", fid->fid, wst->size+2, wst);
 	if (IS_ERR(req)) {
 		err = PTR_ERR(req);
 		goto error;
-- 
1.7.1.1

